### PR TITLE
add mill support

### PR DESCRIPTION
**Pull request by sequencer**

### CODE REVIEW

**Comment by terpstra** [comment]
@jackkoenig Is this the right thing to be doing?

**Comment by sequencer** [comment]
Here is the example usage to this:
https://github.com/sequencer/rocket-playground/blob/3631287b2cd7c9756765f62ef7c7733d3b564b2a/build.sc#L33
I this can really help cross project dependency build by mill.

File: build-rules/mill/build.sc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import mill._
+import mill.scalalib._
+import mill.scalalib.publish._
+import ammonite.ops._
+
+object config extends config
+
+class config extends ScalaModule with PublishModule {
+  def scalaVersion = "2.12.8"
+  def millSourcePath = pwd / up / up / 'design / 'craft
```
Comment (jackkoenig) [inline build-rules/mill/build.sc:10]: is `pwd` relative to where the user runs the command or relative to this file?

File: build-rules/mill/build.sc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  def publishVersion = "1.0"
+  def pomSettings = PomSettings(
+    description = "A Scala library for Context-Dependent Evironments",
+    organization = "org.chipsalliance",
+    url = "https://github.com/chipsalliance/api-config-chipsalliance",
+    licenses = Seq(License.`Apache-2.0`),
+    versionControl = VersionControl.github("chipsalliance", "api-config-chipsalliance"),
+    developers = Seq()
+  )
```
Comment (jackkoenig) [inline build-rules/mill/build.sc:19]: Is this necessary to be a `PublishModule`? I don't think we should publish this from `mill`, in particular picking the version `1.0`.

File: build-rules/mill/build.sc (hunk 1 above)
Comment (sequencer) [reply build-rules/mill/build.sc:10]: Yep. More clearly, I use `os.pwd` and `os.up` now.

File: build-rules/mill/build.sc (hunk 2 above)
Comment (sequencer) [reply build-rules/mill/build.sc:19]: My bad removed.

**Pull request merged**